File: app/src/main/java/com/zaze/demo/model/impl/DeviceModelImpl.java
package com.zaze.demo.model.impl;

import android.app.ActivityManager;
import android.os.Build;
import android.text.TextUtils;

import com.zaze.demo.app.MyApplication;
import com.zaze.demo.model.DeviceModel;
import com.zaze.demo.model.entity.DeviceStatus;
import com.zaze.utils.FileUtil;
import com.zaze.utils.ZCommand;
import com.zaze.utils.ZDeviceUtil;
import com.zaze.utils.ZNetUtil;
import com.zaze.utils.ZStringUtil;
import com.zaze.utils.date.DateUtil;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.TimeZone;

/**
 * Description :
 *
 * @author : ZAZE
 * @version : 2017-01-22 - 17:26
 */
public class DeviceModelImpl implements DeviceModel {
    @Override
    public ArrayList<DeviceStatus> getDeviceInfo() {
        ArrayList<DeviceStatus> list = new ArrayList<>();
        DeviceStatus deviceStatus;
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("Android版本");
        deviceStatus.setName("os");
        deviceStatus.setContent(Build.VERSION.RELEASE);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("androidSDK");
        deviceStatus.setName("sdkInt");
        deviceStatus.setContent(String.valueOf(Build.VERSION.SDK_INT));
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("型号");
        deviceStatus.setName("model");
        deviceStatus.setContent(Build.MODEL);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("主板");
        deviceStatus.setName("board");
        deviceStatus.setContent(Build.BOARD);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("版本号");
        deviceStatus.setName("display");
        deviceStatus.setContent(Build.DISPLAY);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("FINGERPRINT");
        deviceStatus.setName("fingerprint");
        deviceStatus.setContent(Build.FINGERPRINT);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("基带版本");
        deviceStatus.setName("bootloader");
        deviceStatus.setContent(Build.BOOTLOADER);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("系统定制商");
        deviceStatus.setName("brand");
        deviceStatus.setContent(Build.BRAND);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("硬件序列号");
        deviceStatus.setContent(Build.SERIAL);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("Product");
        deviceStatus.setName("product");
        deviceStatus.setContent(Build.PRODUCT);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("描述Build的标签");
        deviceStatus.setContent(Build.TAGS);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("TIME");
        deviceStatus.setContent(DateUtil.timeMillisToString(Build.TIME, "yyyy-MM-dd HH:mm:ss", TimeZone.getDefault()));
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("builder类型");
        deviceStatus.setContent(Build.TYPE);
        list.add(deviceStatus);
        // --------------------------------------------------
//        deviceStatus = new DeviceStatus();
//        deviceStatus.setTag("内核版本");
//        deviceStatus.setContent(FileUtil.readFromFile(FileUtil.getSDCardRoot() + "/proc/version").toString());
//        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("USER");
        deviceStatus.setContent(Build.USER);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("ROOT");
        deviceStatus.setName("root");
        deviceStatus.setContent(String.valueOf(ZCommand.isRoot()));
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("设备号");
        deviceStatus.setName("deviceId");
        deviceStatus.setContent(ZDeviceUtil.getUUID(MyApplication.getInstance()));
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("IMEI");
        deviceStatus.setName("IMEI");
        deviceStatus.setContent(ZDeviceUtil.getDeviceId(MyApplication.getInstance()));
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("CPU 数目");
        deviceStatus.setContent(Runtime.getRuntime().availableProcessors() + "");
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("cpu指令集");
        deviceStatus.setContent(Build.CPU_ABI);
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("cpu指令集2");
        deviceStatus.setContent(Build.CPU_ABI2);
        list.add(deviceStatus);
        // --------------------------------------------------
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            deviceStatus = new DeviceStatus();
            deviceStatus.setTag("cpu指令集");
            deviceStatus.setContent(Arrays.toString(Build.SUPPORTED_ABIS));
            list.add(deviceStatus);
        }
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("运存情况");
        String vmMax = ZStringUtil.format("%sMB", ZDeviceUtil.getRuntimeMaxMemory() >> 20);
        String vmTotal = ZStringUtil.format("%sMB", ZDeviceUtil.getRuntimeTotalMemory() >> 20);
        String vmFree = ZStringUtil.format("%sKB", ZDeviceUtil.getRuntimeFreeMemory() >> 10);
        deviceStatus.setContent(ZStringUtil.format("最大运存 : %s\n总运存 : %s\n剩余运存 : %s", vmMax, vmTotal, vmFree));
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("内存空间");
        ActivityManager.MemoryInfo memoryInfo = ZDeviceUtil.getDeviceMemory(MyApplication.getInstance());
        String deviceTotalMem = ZStringUtil.format("%sMB", memoryInfo.totalMem >> 20);
        String deviceAvailMem = ZStringUtil.format("%sMB", memoryInfo.availMem >> 20);
        deviceStatus.setContent(ZStringUtil.format("总内存 : %s\n可用内存 : %s", deviceTotalMem, deviceAvailMem));
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("SD卡空间状况");
        String sdTotal = ZStringUtil.format("%sMB", ZDeviceUtil.getSdTotalSpace() >> 20);
        String sdFree = ZStringUtil.format("%sMB", ZDeviceUtil.getSdFreeSpace() >> 20);
        deviceStatus.setContent(ZStringUtil.format("总空间 : %s\n剩余空间 : %s", sdTotal, sdFree));
        list.add(deviceStatus);
        // --------------------------------------------------
        deviceStatus = new DeviceStatus();
        deviceStatus.setTag("机身空间状况");
        String dataTotal = ZStringUtil.format("%sMB", ZDeviceUtil.getDataTotalSpace() >> 20);
        String dataFree = ZStringUtil.format("%sMB", ZDeviceUtil.getDataFreeSpace() >> 20);
        deviceStatus.setContent(ZStringUtil.format("总大小 : %s\n剩余大小 : %s", dataTotal, dataFree));
        list.add(deviceStatus);
        // --------------------------------------------------


        // --------------------------------------------------
        JSONObject jsonObject = new JSONObject();
        try {
            for (DeviceStatus status : list) {
                String name = status.getName();
                if (!TextUtils.isEmpty(name)) {
                    jsonObject.put(name, status.getContent());
                }
            }
            jsonObject.put("mac", ZNetUtil.getWLANMacAddress());
        } catch (JSONException e) {
            e.printStackTrace();
        }
        FileUtil.writeToFile("/sdcard/aaa.json", jsonObject.toString(), false);

//        jsonObject.put("deviceId", XhApplication.getInstance().getDeviceId());
//        jsonObject.put("model", Build.MODEL);
//        jsonObject.put("os", Build.VERSION.RELEASE);
//        jsonObject.put("imei", ZStringUtil.parseString(ZDeviceUtil.getDeviceId(XhApplication.getInstance())));
//        jsonObject.put("mac", NetworkUtil.getWLANMacAddress());


        return list;
    }
}

File: app/src/main/java/com/zaze/demo/model/impl/DeviceModelImpl.kt
package com.zaze.demo.model.impl

import android.net.wifi.WifiManager
import android.os.Build
import android.text.TextUtils
import com.zaze.common.base.BaseApplication
import com.zaze.demo.app.MyApplication
import com.zaze.demo.model.DeviceModel
import com.zaze.demo.model.entity.DeviceStatus
import com.zaze.demo.util.StorageLoader
import com.zaze.utils.*
import com.zaze.utils.FileUtil.writeToFile
import com.zaze.utils.date.DateUtil
import org.json.JSONException
import org.json.JSONObject
import java.lang.reflect.Method
import java.util.*

/**
 * Description :
 *
 * @author : ZAZE
 * @version : 2017-01-22 - 17:26
 */
class DeviceModelImpl : DeviceModel {
    override fun getDeviceInfo(): ArrayList<DeviceStatus> {
        val list = ArrayList<DeviceStatus>().apply {
            val displayProfile = DisplayUtil.getDisplayProfile()
            add(
                DeviceStatus(
                    tag = "屏幕分辨率",
                    content = "${displayProfile.realWidthPixels}x${displayProfile.realHeightPixels}(${displayProfile.widthDp}x${displayProfile.heightDp})"
                )
            )
            add(
                DeviceStatus(
                    tag = "屏幕密度",
                    content = "${displayProfile.getDensityDpiName()}(${displayProfile.densityDpi})(${displayProfile.density})"
                )
            )
            add(
                DeviceStatus(
                    tag = "WLAN MAC地址",
                    content = NetUtil.getWLANMacAddress()
                )
            )
            add(
                DeviceStatus(
                    tag = "Android版本",
                    name = "os",
                    content = Build.VERSION.RELEASE
                )
            )
            add(
                DeviceStatus(
                    tag = "android SDK",
                    name = "sdkInt",
                    content = Build.VERSION.SDK_INT.toString()
                )
            )
            add(
                DeviceStatus(
                    tag = "型号",
                    name = "model",
                    content = Build.MODEL
                )
            )
            add(
                DeviceStatus(
                    tag = "主板",
                    name = "board",
                    content = Build.BOARD
                )
            )
            add(
                DeviceStatus(
                    tag = "版本号",
                    name = "display",
                    content = Build.DISPLAY
                )
            )
            add(
                DeviceStatus(
                    tag = "FINGERPRINT",
                    name = "fingerprint",
                    content = Build.FINGERPRINT
                )
            )
            add(
                DeviceStatus(
                    tag = "基带版本",
                    name = "bootloader",
                    content = Build.BOOTLOADER
                )
            )
            add(
                DeviceStatus(
                    tag = "系统定制商",
                    name = "brand",
                    content = Build.BRAND
                )
            )
            add(
                DeviceStatus(
                    tag = "硬件序列号",
                    content = Build.SERIAL
                )
            )
            add(
                DeviceStatus(
                    tag = "Product",
                    name = "product",
                    content = Build.PRODUCT
                )
            )
            add(
                DeviceStatus(
                    tag = "描述Build的标签",
                    name = "tags",
                    content = Build.TAGS
                )
            )
            add(
                DeviceStatus(
                    tag = "TIME",
                    name = "tags",
                    content = DateUtil.timeMillisToString(
                        Build.TIME,
                        "yyyy-MM-dd HH:mm:ss",
                        TimeZone.getDefault()
                    )
                )
            )
            add(
                DeviceStatus(
                    tag = "builder类型",
                    name = "type",
                    content = Build.TYPE
                )
            )
            add(
                DeviceStatus(
                    tag = "USER",
                    name = "user",
                    content = Build.USER
                )
            )
            add(
                DeviceStatus(
                    tag = "ROOT",
                    name = "root",
                    content = ZCommand.isRoot().toString()
                )
            )
            add(
                DeviceStatus(
                    tag = "设备号",
                    name = "deviceId",
                    content = DeviceUtil.getUUID(MyApplication.getInstance())
                )
            )
            add(
                DeviceStatus(
                    tag = "IMEI",
                    name = "imei",
                    content = DeviceUtil.getDeviceId(MyApplication.getInstance())
                )
            )
            // --------------------------------------------------

            val result = ZCommand.execCmdForRes("cat /proc/version")
            add(
                DeviceStatus(
                    tag = "内核版本",
                    name = "kernel",
                    content = if (result.isSuccess) {
                        result.successMsg
                    } else {
                        result.errorMsg
                    }
                )
            )
            add(
                DeviceStatus(
                    tag = "CPU 数目",
                    content = Runtime.getRuntime().availableProcessors().toString()
                )
            )
            add(
                DeviceStatus(
                    tag = "cpu指令集",
                    content = Build.CPU_ABI
                )
            )
            add(
                DeviceStatus(
                    tag = "cpu指令集2",
                    content = Build.CPU_ABI2
                )
            )
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                add(
                    DeviceStatus(
                        tag = "cpu指令集",
                        content = Arrays.toString(Build.SUPPORTED_ABIS)
                    )
                )
            }
            // --------------------------------------------------
            add(
                DeviceStatus(
                    tag = "运存情况",
                    content = "最大运存: ${DescriptionUtil.toByteUnit(DeviceUtil.getRuntimeMaxMemory())}\n" +
                            "总运存: ${DescriptionUtil.toByteUnit(DeviceUtil.getRuntimeTotalMemory())}\n" +
                            "剩余运存: ${DescriptionUtil.toByteUnit(DeviceUtil.getRuntimeFreeMemory())}"
                )
            )
            val memoryInfo = DeviceUtil.getDeviceMemory(MyApplication.getInstance())
            add(
                DeviceStatus(
                    tag = "内存空间",
                    content = "总内存: ${DescriptionUtil.toByteUnit(memoryInfo.totalMem)}\n" +
                            "可用内存: ${DescriptionUtil.toByteUnit(memoryInfo.availMem)}"
                )
            )
//            add(
//                DeviceStatus(
//                    tag = "SD卡存储空间",
//                    content = "总空间: ${DescriptionUtil.toByteUnit(DeviceUtil.getSdTotalSpace())}\n" +
//                            "剩余空间: ${DescriptionUtil.toByteUnit(DeviceUtil.getSdFreeSpace())}"
//                )
//            )
            val storageInfo = StorageLoader.loadStorageStats(BaseApplication.getInstance())
            add(
                DeviceStatus(
                    tag = "存储空间",
                    content = "总大小: ${DescriptionUtil.toByteUnit(storageInfo.totalBytes)}\n" +
                            "剩余大小: ${DescriptionUtil.toByteUnit(storageInfo.freeBytes)}"
                )
            )
        }
        // --------------------------------------------------


        // --------------------------------------------------
        val jsonObject = JSONObject()
        try {
            for (status in list) {
                val name = status.name
                if (!TextUtils.isEmpty(name)) {
                    jsonObject.put(name, status.content)
                }
            }
            jsonObject.put("mac", NetUtil.getWLANMacAddress())
        } catch (e: JSONException) {
            e.printStackTrace()
        }
        writeToFile("/sdcard/aaa.json", jsonObject.toString(), false)

//        jsonObject.put("deviceId", XhApplication.getInstance().getDeviceId());
//        jsonObject.put("model", Build.MODEL);
//        jsonObject.put("os", Build.VERSION.RELEASE);
//        jsonObject.put("imei", ZStringUtil.parseString(ZDeviceUtil.getDeviceId(XhApplication.getInstance())));
//        jsonObject.put("mac", NetworkUtil.getWLANMacAddress());
        return list
    }
}